pkg/storage/allocator.go

```diff
@@ -415,6 +415,7 @@ func (a *Allocator) TransferLeaseTarget(
 	stats *replicaStats,
 	checkTransferLeaseSource bool,
 	checkCandidateFullness bool,
+	wantScatter bool,
```

Comment:
I think you can just reuse `checkTransferLeaseSource` for this, which effectively instructs this function to not even consider keeping the lease on the local replica.